vector/dict.go

```diff
@@ -17,6 +19,14 @@ func NewDict(vals Any, index []byte, counts []uint32, nulls *Bool) *Dict {
 	return &Dict{vals, index, counts, nulls}
 }
 
+func NewDictBuildCounts(vals Any, index []byte, nulls *Bool) *Dict {
+	counts := make([]uint32, vals.Len())
+	for _, slot := range index {
+		counts[slot]++
+	}
+	return &Dict{vals, index, counts, nulls}
+}
```

Comment:
Instead of a new constructor, how about changing NewDict to build counts if that parameter is nil?

```go
func NewDict(vals Any, index []byte, counts []uint32, nulls *Bool) *Dict {
	if counts == nil {
		counts = make([]uint32, vals.Len())
		for _, slot := range index {
			counts[slot]++
		}
	}
	return &Dict{vals, index, counts, nulls}
}
```

Reply:
On second thought, since it looks like you want this solely for TestDictDeleteEntries, I don't think we should add it. Just use a composite literal in the test or build the slice there.

Reply:
I also need to know the indexes that we dropped so I can construct the error view.

Reply:
> On second thought, since it looks like you want this solely for TestDictDeleteEntries, I don't think we should add it. Just use a composite literal in the test or build the slice there.

I don't follow this. I want this because I use it network_of for intMastFast

Reply:
Fair

Reply:
I do think we're going to want to have this in `NewDict` as you suggested but we'll wait for that day.

Reply:
Actually I'm being totally silly about this. We don't want DeleteEntries here of ips we want to create a new nets dictionary. I'm reworking this and this will be removed.

Reply:
ok fixed.